llvm/docs/CASObjectFormats/CASObjectFormats.md

```diff
@@ -0,0 +1,78 @@
+# CAS Object (File) Formats
+
+## Objects supported by the CAS
+
+There are three types of objects in LLVMCAS:
+- Blob: opaque sequence of bytes (`StringRef`).
+- Tree: filesystem tree, mapping names to other CAS objects. Similar to a tree
+  in Git, and not particularly relevant here.
+- Node: generic node, referencing any number of CAS objects plus an opaque
+  sequence of bytes.
+
+The object format schemas mostly use `Node`s.
+
+- The top-level `Node` reserves its first CAS reference for a "type-id". This
+  is DAG of CAS objects that identifies the schema.
+- Currently, the kind for other nodes is in their first byte of data. This is
+  a convenience for debugging / dumping them.
+
+## Schemas
+
+See documentation for specifics of schemas:
+
+- NestedV1.md (a nested CAS-based object format)
+- FlatV1.md (a flat CAS-based object format)
+
+There is also a separate document for some common encodings:
+
+- CommonEncodings.md
+
+### Split up descriptor sections
+
+These formats/schemas are optimized for when mega-sections like `__eh_frame`
+and `__compact_unwind` are split up into separate blocks/atoms, with anonymous
+symbols created for each record.
+
+### Split up string pools
+
+Similarly, string sections should be split up with an anonymous symbol for each
+string.
+
+## Major caveats
+
+There are major caveats that prevent the current formats from being useful in
+from being useful in practice. These mostly stem from them handling
+serialization to/from `jitlink::LinkGraph` and not much else. That said,
+they're useful for experimenting with encoding schemes and demonstrating
+storage optimization headroom.
+
+Not clear whether we'll evolve one of these or create a new one.
+
+### Caveat: Does not support real algorithms
```

Comment:
This feels more like just we don't have an interface that compiler or linker can call into.
LinkGraph totally supports linking concepts and FlatV1 can totally do a fair amount of lazy loading if needed. We just don't have API for those.

Reply:
Rewored to "Not ready to support" and clarified text a bit.